src/main/java/org/folio/circulation/domain/StoredRequestRepresentation.java

```diff
@@ -0,0 +1,86 @@
+package org.folio.circulation.domain;
+
+import static java.util.Objects.isNull;
+import static org.folio.circulation.support.JsonPropertyWriter.write;
+
+import java.lang.invoke.MethodHandles;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.json.JsonObject;
+
+class StoredRequestRepresentation {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  JsonObject storedRequest(Request request) {
+    final JsonObject representation = request.asJson();
+
+    addStoredItemProperties(representation, request.getItem());
+    addStoredRequesterProperties(representation, request.getRequester());
+    addStoredProxyProperties(representation, request.getProxy());
+
+    removeDeliveryAddress(representation);
+
+    return representation;
+  }
+
+  private static void addStoredItemProperties(JsonObject request, Item item) {
+    if (item == null || item.isNotFound()) {
+      logUnableAddItemToTheRequest(request, item);
+      return;
+    }
+
+    JsonObject itemSummary = new JsonObject();
+
+    write(itemSummary, "title", item.getTitle());
+    write(itemSummary, "barcode", item.getBarcode());
+
+    request.put("item", itemSummary);
+  }
+
+  private static void addStoredRequesterProperties(JsonObject request, User requester) {
+    if (requester == null) {
+      String msg = "Unable to add requester properties to the request: {}, requester is null.";
+      log.info(msg, request.getString("id"));
+      return;
+    }
+
+    request.put("requester", storedUserSummary(requester));
+  }
+
+  private static void addStoredProxyProperties(JsonObject request, User proxy) {
+    if (proxy == null) {
+      String msg = "Unable to add proxy properties to request {}, proxy object is null";
+      log.info(msg, request.getString("id"));
+      return;
+    }
+
+    request.put("proxy", storedUserSummary(proxy));
+  }
+
+  private static JsonObject storedUserSummary(User user) {
+    JsonObject userSummary = new JsonObject();
+
+    write(userSummary, "lastName", user.getLastName());
+    write(userSummary, "firstName", user.getFirstName());
+    write(userSummary, "middleName", user.getMiddleName());
+    write(userSummary, "barcode", user.getBarcode());
+    return userSummary;
+  }
+
+  private static void logUnableAddItemToTheRequest(JsonObject request, Item item) {
+    String reason = isNull(item) ? "null" : "not found";
+    String msg = "Unable to add item properties to the request: {}, item is {}";
+    log.info(msg, request.getString("id"), reason);
+  }
+
+  private static void removeDeliveryAddress(JsonObject requestRepresentation) {
+    if (requestRepresentation == null) {
```

Comment:
This block is not covered according to Sonar. Does it make sense to add a test that will cover this case? All the other `if (something == null)` blocks are covered in the other methods.

Reply:
I’ll take a look at that, it is probably being highlighted because I split the classes.

I’ve been tempted to remove those checks across the board if I can think of better protection